docs/specs/tsd-008-probe-on-startup.md

```diff
@@ -0,0 +1,55 @@
+# `probe` Option For `startup_error_behavior`
+
+## Objective
+
+Creation of a scheme that allows Telegraf to probe a plugin during its initialization phase to enable enhanced plugin error detection due to configuration or environmental problems.
+
+## Keywords
+
+inputs, outputs, startup, probe, error, ignore, behavior
+
+## Overview
+
+When plugins are first instantiated, Telegraf will call the plugin's `Start()` method (for inputs) or `Connect()` (for outputs) which will initialize its configuration based off of config options and the running environment. It is sometimes the case that while the initialization step succeeds, the upstream service in which the plugin relies on is not actually running, or is not capable of being communicated with due to incorrect configuration or environmental problems. In situations like this, Telegraf does not detect that the plugin's upstream service is not functioning properly, and thus it will continually call the plugin during each `Gather()` iteration. This often has the effect of polluting journald and system logs with voluminous error messages, which creates issues for system administrators who rely on such logs to identify other unrelated system problems.
+
+More background discussion on this option, including other possible avenues, can be viewed [here](https://github.com/influxdata/telegraf/issues/16028).
+
+## Probing
+
+First, we must define what it means for a plugin to `Probe`. Probing is an action whereby the plugin will attempt to communicate with its external service, device, entity, or executable as if it were gathering real metrics. The payload of this communication attempt is not used to record any metrics. The result of a `Probe` shall be an error that indicates whether the communication was successful or not.
+
+We define an interface for this action:
+
+```go
+type Prober interface {
+    Probe() error
+}
+```
+
+In almost all respects, a `Probe` is similar to a `Gather`, the only difference being that `Probe` does not record any metrics and only returns information on the success or failure of the action.
+
+## Configuration
+
+The Telegraf project has already introduced the `startup_error_behavior` configuration which allows the user to define how Telegraf should behave if the plugin fails to start. The `ignore` value allows the plugin to be ignored if the `Start()`/`Connect()` method returns an error. We propose to add an additional value to this parameter called `probe`, that will behave as a superset of the `ignore` behavior. When `startup_error_behavior=probe`, Telegraf will perform the following steps on plugin startup:
```

Comment:
I'm not sure if I totally understand the question. Is your question asking if we can just enable probing for any plugin that implements it, without needing a new config parameter? Personally, I tend to lean towards the end of giving the user _more_ control over what happens, not less. In the case of output plugins, it may very well be the case that users emphatically don't want the plugin probing on startup because that means, probably, that you're sending some fake metrics out somewhere.